pandas/_libs/parsers.pyx

```diff
@@ -1049,12 +1045,12 @@ cdef class TextReader:
             # don't try to upcast EAs
             if (
                 na_count > 0 and not is_extension_array_dtype(col_dtype)
-                or self.use_nullable_dtypes
+                or self.dtype_backend != "numpy"
             ):
-                use_nullable_dtypes = self.use_nullable_dtypes and col_dtype is None
+                use_dtype_backend = self.dtype_backend != "numpy" and col_dtype is None
```

Comment:
Not sure if the previous variable name `use_nullable_dtypes` is still more appropriate here. No big deal, up to you.

Reply:
As long as no_default is used for numpy I like this more